=== modified file 'report_process_production/report/process_production_report.rml'
--- report_process_production/report/process_production_report.rml	2012-10-18 23:20:04 +0000
+++ report_process_production/report/process_production_report.rml	2012-10-18 23:20:27 +0000
@@ -2,6 +2,19 @@
 <document filename="test.pdf">
   <template pageSize="(612.0,792.0)" title="Test" author="Martin Simon" allowSplitting="20">
     <pageTemplate id="first">
+    <pageGraphics>
+        <setFont name="Courier-Bold" size="12.0"/>
+        <drawCentredString x="10.5cm" y="0.8cm">Página: <pageNumber/></drawCentredString>
+    <place x="18.0cm"  y="25.0cm"  width="90"  height="60">
+      <blockTable colWidths="85" style="Table_order">
+        <tr>
+          <td>
+            <para style="P7">Impreso: [[time.strftime('%d-%b-%Y')]]</para>
+          </td>
+        </tr>
+      </blockTable>
+    </place>
+        </pageGraphics>
       <frame id="first" x1="28.0" y1="57.0" width="556" height="678"/>
     </pageTemplate>
   </template>
@@ -10,6 +23,11 @@
       <blockAlignment value="LEFT"/>
       <blockValign value="TOP"/>
     </blockTableStyle>
+    <blockTableStyle id="Table_order">
+      <blockAlignment value="LEFT"/>
+      <blockValign value="TOP"/>
+      <lineStyle kind="GRID" colorName="#e6e6e6" start="0,0" stop="-1,-1"/>
+    </blockTableStyle>
     <blockTableStyle id="Table2">
       <blockAlignment value="LEFT"/>
       <blockValign value="TOP"/>
@@ -35,19 +53,19 @@
       <blockValign value="TOP"/>
       <lineStyle kind="LINEBEFORE" colorName="#000000" start="0,0" stop="0,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="0,0" stop="0,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="0,0" stop="0,0"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="0,0" stop="0,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="0,-1" stop="0,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="1,0" stop="1,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="1,0" stop="1,0"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="1,0" stop="1,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="1,-1" stop="1,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="2,0" stop="2,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="2,0" stop="2,0"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="2,0" stop="2,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="2,-1" stop="2,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="3,0" stop="3,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="3,0" stop="3,0"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="3,0" stop="3,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="3,-1" stop="3,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="4,0" stop="4,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="4,0" stop="4,0"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="4,0" stop="4,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="4,-1" stop="4,-1"/>
     </blockTableStyle>
     <blockTableStyle id="Table3">
@@ -75,23 +93,23 @@
       <lineStyle kind="LINEBELOW" colorName="#000000" start="4,-1" stop="4,-1"/>
       <lineStyle kind="LINEBEFORE" colorName="#000000" start="0,1" stop="0,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="0,1" stop="0,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="0,1" stop="0,1"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="0,1" stop="0,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="0,-1" stop="0,-1"/>
       <lineStyle kind="LINEBEFORE" colorName="#000000" start="1,1" stop="1,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="1,1" stop="1,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="1,1" stop="1,1"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="1,1" stop="1,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="1,-1" stop="1,-1"/>
       <lineStyle kind="LINEBEFORE" colorName="#000000" start="2,1" stop="2,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="2,1" stop="2,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="2,1" stop="2,1"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="2,1" stop="2,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="2,-1" stop="2,-1"/>
       <lineStyle kind="LINEBEFORE" colorName="#000000" start="3,1" stop="3,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="3,1" stop="3,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="3,1" stop="3,1"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="3,1" stop="3,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="3,-1" stop="3,-1"/>
       <lineStyle kind="LINEBEFORE" colorName="#000000" start="4,1" stop="4,-1"/>
       <lineStyle kind="LINEAFTER" colorName="#000000" start="4,1" stop="4,-1"/>
-      <lineStyle kind="LINEABOVE" colorName="#000000" start="4,1" stop="4,1"/>
+      <lineStyle kind="LINEABOVE" colorName="#000000" start="4,1" stop="4,-1"/>
       <lineStyle kind="LINEBELOW" colorName="#000000" start="4,-1" stop="4,-1"/>
     </blockTableStyle>
     <initialize>
@@ -100,17 +118,18 @@
     <paraStyle name="P1" fontName="Helvetica" fontSize="6.0" leading="8" alignment="CENTER"/>
     <paraStyle name="P2" fontName="Helvetica-Bold" fontSize="9.0" leading="11" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
     <paraStyle name="P3" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P4" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P5" fontName="Helvetica" fontSize="6.0" leading="8" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P6" fontName="Helvetica-Bold" fontSize="4.0" leading="5" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P7" fontName="Helvetica" fontSize="8.0" leading="10" alignment="LEFT"/>
-    <paraStyle name="P8" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="CENTER"/>
+    <paraStyle name="P4" fontName="Helvetica" fontSize="6.0" leading="8" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P5" fontName="Helvetica-Bold" fontSize="4.0" leading="5" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P6" fontName="Helvetica" fontSize="8.0" leading="10" alignment="LEFT"/>
+    <paraStyle name="P7" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="CENTER"/>
+    <paraStyle name="P8" fontName="Helvetica" fontSize="8.0" leading="10" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
     <paraStyle name="P9" fontName="Helvetica" fontSize="8.0" leading="10" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P10" fontName="Helvetica" fontSize="8.0" leading="10" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P11" fontName="Helvetica" fontSize="8.0" leading="10" alignment="RIGHT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P12" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P13" fontName="Helvetica" fontSize="6.0" leading="8" alignment="LEFT" spaceBefore="0.0" spaceAfter="0.0"/>
-    <paraStyle name="P14" fontName="Helvetica-Bold" fontSize="10.0" leading="13" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P10" fontName="Helvetica" fontSize="8.0" leading="10" alignment="RIGHT" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P11" fontName="Helvetica" fontSize="6.0" leading="8" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P12" fontName="Helvetica-Bold" fontSize="10.0" leading="13" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P13" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P14" fontName="Helvetica-Bold" fontSize="8.0" leading="10" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
+    <paraStyle name="P15" fontName="Helvetica-Bold" fontSize="10.0" leading="13" alignment="CENTER" spaceBefore="0.0" spaceAfter="0.0"/>
     <paraStyle name="Standard" fontName="Helvetica"/>
     <paraStyle name="Heading" fontName="Helvetica" fontSize="14.0" leading="17" spaceBefore="12.0" spaceAfter="6.0"/>
     <paraStyle name="Text body" fontName="Helvetica" spaceBefore="0.0" spaceAfter="6.0"/>
@@ -125,123 +144,123 @@
   </stylesheet>
   <story>
     <section>
-      <para style="P4">[[ repeatIn(get_sin(data['form']), 's') ]]</para>
-      <para style="P14">Reporte de proceso de producción</para>
+      <para style="P13">[[ repeatIn(get_sin(data['form']), 's') ]]</para>
+      <para style="P12">Reporte de proceso de producción</para>
       <blockTable colWidths="242.0,61.0,66.0,122.0,62.0" style="Table2">
         <tr>
           <td>
-            <para style="P8">[[ s['name'] ]]</para>
-          </td>
-          <td>
-            <para style="P8">[[ s['qty'] ]]</para>
-          </td>
-          <td>
-            <para style="P8">[[ s['uom'] ]]</para>
-          </td>
-          <td>
-            <para style="P8">[[ s['categ'] ]]</para>
-          </td>
-          <td>
-            <para style="P8">[[ s['real'] ]]</para>
+            <para style="P7">[[ s['name'] ]]</para>
+          </td>
+          <td>
+            <para style="P7">[[ s['qty'] ]]</para>
+          </td>
+          <td>
+            <para style="P7">[[ s['uom'] ]]</para>
+          </td>
+          <td>
+            <para style="P7">[[ s['categ'] ]]</para>
+          </td>
+          <td>
+            <para style="P7">[[ s['real'] ]]</para>
           </td>
         </tr>
       </blockTable>
       <para style="P1">
         <font color="white"> </font>
       </para>
-      <para style="P6">
+      <para style="P5">
         <font color="white"> </font>
       </para>
     </section>
     <section>
-      <para style="P9">[[ repeatIn(get_table(data['form'],objects),'o') ]]</para>
+      <para style="P8">[[ repeatIn(get_table(data['form'],objects),'o') ]]</para>
       <para style="P3">Producción: [[ o.name ]] , Producto: [[o.product_id.name ]], Codigo: [[o.product_id.default_code ]] , Cantidad: [[ o.product_qty ]] [[o.product_uom.name ]] , Fecha Planificada: [[o.date_planned or '']] , Lista De Materiales: [[ o.bom_id and o.bom_id.name or '' ]]</para>
       <blockTable colWidths="242.0,61.0,66.0,121.0,62.0" style="Table1">
         <tr>
           <td>
-            <para style="P10">[[ repeatIn(get_production(data['form'], o.id ), 'p' ) ]]</para>
-            <para style="P10">[[ p.product_id.default_code]]:: [[ p.product_id.name ]]</para>
-          </td>
-          <td>
-            <para style="P11">[[ formatLang(p.product_qty, get_number_dg()) ]]</para>
-          </td>
-          <td>
-            <para style="P10">[[ p.product_uom.name ]]</para>
-          </td>
-          <td>
-            <para style="P10">[[ p.product_id.categ_id.name ]]</para>
-          </td>
-          <td>
-            <para style="P7">
+            <para style="P9">[[ repeatIn(get_production(data['form'], o.id ), 'p' ) ]]</para>
+            <para style="P9">[[ p.product_id.default_code]]:: [[ p.product_id.name ]]</para>
+          </td>
+          <td>
+            <para style="P10">[[ formatLang(p.product_qty, get_number_dg()) ]]</para>
+          </td>
+          <td>
+            <para style="P9">[[ p.product_uom.name ]]</para>
+          </td>
+          <td>
+            <para style="P9">[[ p.product_id.categ_id.name ]]</para>
+          </td>
+          <td>
+            <para style="P6">
               <font color="white"> </font>
             </para>
           </td>
         </tr>
       </blockTable>
-      <para style="P5">
+      <para style="P4">
         <font color="white"> </font>
       </para>
     </section>
-    <para style="P5">
+    <para style="P4">
       <font color="white"> </font>
     </para>
-    <para style="P13">______________________________________________________________________________________________________________________________________________________________________</para>
-    <para style="P13">
+    <para style="P11">______________________________________________________________________________________________________________________________________________________________________</para>
+    <para style="P11">
       <font color="white"> </font>
     </para>
     <section>
-      <para style="P12">[[ repeatIn(get_group(data['form']), 'g') ]]</para>
-      <para style="P14">Reporte de proceso de producción agrupado</para>
+      <para style="P14">[[ repeatIn(get_group(data['form']), 'g') ]]</para>
+      <para style="P12">Reporte de proceso de producción agrupado</para>
       <para style="P2">
         <font color="white"> </font>
       </para>
       <blockTable colWidths="241.0,63.0,66.0,121.0,61.0" style="Table3">
         <tr>
           <td>
-            <para style="P8">Nombre</para>
-          </td>
-          <td>
-            <para style="P8">Cantidad</para>
-          </td>
-          <td>
-            <para style="P8">UOM</para>
-          </td>
-          <td>
-            <para style="P8">Categoria</para>
-          </td>
-          <td>
-            <para style="P8">Real</para>
+            <para style="P7">Nombre</para>
+          </td>
+          <td>
+            <para style="P7">Cantidad</para>
+          </td>
+          <td>
+            <para style="P7">UOM</para>
+          </td>
+          <td>
+            <para style="P7">Categoria</para>
+          </td>
+          <td>
+            <para style="P7">Real</para>
           </td>
         </tr>
         <tr>
           <td>
-            <para style="P10">[[ repeatIn(get_production_group(data['form'] ), 'p' ) ]]</para>
-            <para style="P10">[[ p['code'] ]]::[[ p['name'] ]]</para>
-          </td>
-          <td>
-            <para style="P11">[[ formatLang(p['product_qty'], get_number_dg()) ]]</para>
-          </td>
-          <td>
-            <para style="P10">[[ p['product_uom'] ]]</para>
-          </td>
-          <td>
-            <para style="P10">[[ p['product_categ'] ]]</para>
-          </td>
-          <td>
-            <para style="P7">
+            <para style="P9">[[ repeatIn(get_production_group(data['form'] ), 'p' ) ]]</para>
+            <para style="P9">[[ p['code'] ]]::[[ p['name'] ]]</para>
+          </td>
+          <td>
+            <para style="P10">[[ formatLang(p['product_qty'], get_number_dg()) ]]</para>
+          </td>
+          <td>
+            <para style="P9">[[ p['product_uom'] ]]</para>
+          </td>
+          <td>
+            <para style="P9">[[ p['product_categ'] ]]</para>
+          </td>
+          <td>
+            <para style="P6">
               <font color="white"> </font>
             </para>
           </td>
         </tr>
       </blockTable>
-      <para style="P5">
+      <para style="P4">
         <font color="white"> </font>
       </para>
-      <para style="P5">
+      <para style="P4">
         <font color="white"> </font>
       </para>
     </section>
-    <para style="P13">
+    <para style="P11">
       <font color="white"> </font>
     </para>
   </story>

