=== modified file 'addons/openerp/controllers/templates/header.mako'
--- addons/openerp/controllers/templates/header.mako	2011-02-17 13:59:18 +0000
+++ addons/openerp/controllers/templates/header.mako	2012-03-04 23:51:47 +0000
@@ -25,7 +25,12 @@
 <td id="top" colspan="3">
     <p id="cmp_logo">
         <a href="/" target="_top">
+    % if not logged:
             <img alt="OpenERP" id="company_logo" src="/openerp/static/images/openerp_small.png"/>
+    % endif
+    % if logged:
+            <img alt="OpenERP" id="company_logo" src="/web_multilogo/static/images/${rpc.session.db}/openerp_small.png"/>
+    % endif
         </a>
     </p>
     % if logged:

=== modified file 'addons/openerp/controllers/templates/login.mako'
--- addons/openerp/controllers/templates/login.mako	2010-12-13 14:24:06 +0000
+++ addons/openerp/controllers/templates/login.mako	2012-03-07 06:08:30 +0000
@@ -31,7 +31,7 @@
 
                     <fieldset class="box">
                         <legend style="padding: 4px;">
-                            <img src="/openerp/static/images/stock/stock_person.png" alt=""/>
+                            <img src="/openerp/static/images/vauxoo_favicon.gif" alt=""/>
                         </legend>
                         <div class="box2" style="padding: 5px 5px 20px 5px">
                             <table width="100%" align="center" cellspacing="2px" cellpadding="0" style="border:none;">
@@ -82,9 +82,11 @@
                     <table cellpadding="0" cellspacing="0" width="100%" style="border:none;">
                         <tr>
                             <td style="padding-left:0;"><h3> ${_("Top Contributor:")}</h3></td>
+                            <td style="padding-left:0;"><h3> ${_("Powered by Vauxoo:")}</h3></td>
                         </tr>
                         <tr>
-                            <td style="padding-left:0;"><img src="/openerp/static/images/axelor_logo.png"/></td>
+                            <td style="padding-left:0;"><a href="http://vauxoo.com"><img src="/openerp/static/images/axelor_logo.png"/></a></td>
+                            <td style="padding-left:0;"><img src="/openerp/static/images/vauxoo_logo.png"/></td>
                         </tr>
                     </table>
                 </div>

