=== modified file 'product/product.py'
--- product/product.py	2012-04-03 13:05:28 +0000
+++ product/product.py	2013-05-09 20:38:53 +0000
@@ -125,7 +125,7 @@
         ('factor_gt_zero', 'CHECK (factor!=0)', 'The conversion ratio for a unit of measure cannot be 0!')
     ]
 
-    def _compute_qty(self, cr, uid, from_uom_id, qty, to_uom_id=False):
+    def _compute_qty(self, cr, uid, from_uom_id, qty, to_uom_id=False, context=None):
         if not from_uom_id or not qty or not to_uom_id:
             return qty
         uoms = self.browse(cr, uid, [from_uom_id, to_uom_id])
@@ -133,14 +133,17 @@
             from_unit, to_unit = uoms[0], uoms[-1]
         else:
             from_unit, to_unit = uoms[-1], uoms[0]
-        return self._compute_qty_obj(cr, uid, from_unit, qty, to_unit)
+        return self._compute_qty_obj(cr, uid, from_unit, qty, to_unit, context=context)
 
     def _compute_qty_obj(self, cr, uid, from_unit, qty, to_unit, context=None):
-        if context is None:
+        product_name = ''
+        if context:
+            product_name = context.get('product_name')
+        else:
             context = {}
         if from_unit.category_id.id <> to_unit.category_id.id:
             if context.get('raise-exception', True):
-                raise osv.except_osv(_('Error !'), _('Conversion from Product UoM %s to Default UoM %s is not possible as they both belong to different Category!.') % (from_unit.name,to_unit.name,))
+                raise osv.except_osv(_('Error !'), _('Conversion from Product UoM %s of product %s to Default UoM %s is not possible as they both belong to different Category!.') % (from_unit.name, product_name, to_unit.name,))
             else:
                 return qty
         amount = qty / from_unit.factor

