=== modified file 'stock/stock.py'
--- stock/stock.py	2012-08-07 10:04:35 +0000
+++ stock/stock.py	2012-09-19 18:43:15 +0000
@@ -2445,6 +2445,9 @@
         #quantity should in MOVE UOM
         if context is None:
             context = {}
+            
+        quantity=self.pool.get('product.uom')._compute_qty(cr, uid, context.get('product_uom',False), quantity, to_uom_id=context.get('product_uom_move',False))
+        
         if quantity <= 0:
             raise osv.except_osv(_('Warning!'), _('Please provide Proper Quantity !'))
         res = []
@@ -2458,13 +2461,14 @@
             if quantity_rest <= 0:
                 quantity_rest = 0
                 uos_qty_rest = 0
-                quantity = move.product_qty
+                quantity = quantity
 
             uos_qty = quantity / move_qty * move.product_uos_qty
             if quantity_rest > 0:
                 default_val = {
-                    'product_qty': quantity,
+                    'product_qty': context.get('quantity',quantity),
                     'product_uos_qty': uos_qty,
+                    'product_uom' : context.get('product_uom',move.product_uom.id),
                     'state': move.state,
                     'location_id': location_id or move.location_id.id,
                 }
@@ -2482,6 +2486,7 @@
                 update_val = {
                         'product_qty' : quantity_rest,
                         'product_uos_qty' : uos_qty_rest,
+                        'product_uom' : move.product_uom.id,
                         'location_id': location_id or move.location_id.id,
                 }
                 self.write(cr, uid, [move.id], update_val)

