=== modified file 'product/pricelist.py'
--- product/pricelist.py	2013-03-21 14:34:59 +0000
+++ product/pricelist.py	2013-04-10 01:08:59 +0000
@@ -154,14 +154,13 @@
         date = time.strftime('%Y-%m-%d')
         if 'date' in context:
             date = context['date']
-
         currency_obj = self.pool.get('res.currency')
         product_obj = self.pool.get('product.product')
         product_category_obj = self.pool.get('product.category')
         product_uom_obj = self.pool.get('product.uom')
         supplierinfo_obj = self.pool.get('product.supplierinfo')
         price_type_obj = self.pool.get('product.price.type')
-
+        product_pricelist_item_obj = self.pool.get('product.pricelist.item')
         # product.pricelist.version:
         if not pricelist_ids:
             pricelist_ids = self.pool.get('product.pricelist').search(cr, uid, [], context=context)
@@ -221,8 +220,9 @@
                         'AND price_version_id = %s '
                         'AND (min_quantity IS NULL OR min_quantity <= %s) '
                         'AND i.price_version_id = v.id AND v.pricelist_id = pl.id '
-                    'ORDER BY sequence',
-                    (tmpl_id, product_id) + partner_args + (pricelist_version_ids[0], qty))
+                        'AND i.date <= %s'
+                        'ORDER BY '+product_pricelist_item_obj._order,
+                    (tmpl_id, product_id) + partner_args + (pricelist_version_ids[0], qty, date))
                 res1 = cr.dictfetchall()
                 uom_price_already_computed = False
                 for res in res1:

