=== modified file 'product/pricelist.py'
--- product/pricelist.py	2011-04-21 07:45:29 +0000
+++ product/pricelist.py	2012-05-30 22:19:35 +0000
@@ -161,6 +161,7 @@
         supplierinfo_obj = self.pool.get('product.supplierinfo')
         price_type_obj = self.pool.get('product.price.type')
         product_pricelist_version_obj = self.pool.get('product.pricelist.version')
+        product_pricelist_item_obj = self.pool.get('product.pricelist.item')
 
         # product.pricelist.version:
         if pricelist_ids:
@@ -209,7 +210,7 @@
                     categ_where = '(categ_id IN (' + ','.join(map(str, categ_ids)) + '))'
                 else:
                     categ_where = '(categ_id IS NULL)'
-
+                
                 cr.execute(
                     'SELECT i.*, pl.currency_id '
                     'FROM product_pricelist_item AS i, '
@@ -220,7 +221,7 @@
                         'AND price_version_id = %s '
                         'AND (min_quantity IS NULL OR min_quantity <= %s) '
                         'AND i.price_version_id = v.id AND v.pricelist_id = pl.id '
-                    'ORDER BY sequence',
+                    'ORDER BY '+product_pricelist_item_obj._order,
                     (tmpl_id, product_id, plversion_ids[0], qty))
                 res1 = cr.dictfetchall()
                 uom_price_already_computed = False

