=== modified file 'account_voucher/account_voucher.py'
--- account_voucher/account_voucher.py	2013-06-07 11:38:29 +0000
+++ account_voucher/account_voucher.py	2013-06-25 01:47:39 +0000
@@ -1181,6 +1181,8 @@
         move_line_obj = self.pool.get('account.move.line')
         currency_obj = self.pool.get('res.currency')
         tax_obj = self.pool.get('account.tax')
+        voucher_line_obj = self.pool.get('account.voucher.line')
+        user_obj = self.pool.get('res.users')
         tot_line = line_total
         rec_lst_ids = []
 
@@ -1209,7 +1211,15 @@
                 sign = voucher.type in ('payment', 'purchase') and -1 or 1
                 currency_rate_difference = sign * (line.move_line_id.amount_residual - amount)
             else:
-                currency_rate_difference = 0.0
+                line_id = line.id
+                move_line_id = voucher_line_obj.browse(cr, uid, line.id, context=context).move_line_id.id
+                date_move = move_line_obj.browse(cr, uid, move_line_id).date
+                ctx = context.copy()
+                ctx.update({'date' : date_move})
+                company_user = user_obj.browse(cr, uid, uid).company_id
+                rate_invoice = currency_obj.compute(cr, uid, line.move_line_id.currency_id.id, company_user.currency_id.id, 1, context=ctx)
+                rate_voucher = currency_obj.compute(cr, uid, line.move_line_id.currency_id.id, company_user.currency_id.id, 1, context=context)
+                currency_rate_difference = (rate_invoice - rate_voucher) * (amount/rate_voucher)
             move_line = {
                 'journal_id': voucher.journal_id.id,
                 'period_id': voucher.period_id.id,

