=== modified file 'mrp/mrp.py'
--- mrp/mrp.py	2012-10-05 10:44:47 +0000
+++ mrp/mrp.py	2012-11-30 16:57:38 +0000
@@ -885,7 +885,7 @@
                     'move_id': shipment_move_id,
                     'company_id': production.company_id.id,
                 })
-        wf_service.trg_validate(uid, procurement_order._name, procurement_id, 'button_confirm', cr)
+        #wf_service.trg_validate(uid, procurement_order._name, procurement_id, 'button_confirm', cr) uncomment to confirm procurements
         return procurement_id
 
     def _make_production_internal_shipment_line(self, cr, uid, production_line, shipment_id, parent_move_id, destination_location_id=False, context=None):

