=== modified file 'procurement/procurement.py'
--- procurement/procurement.py	2012-01-04 06:38:07 +0000
+++ procurement/procurement.py	2012-11-30 19:23:22 +0000
@@ -103,7 +103,7 @@
             " a make to order method."),
 
         'note': fields.text('Note'),
-        'message': fields.char('Latest error', size=64, help="Exception occurred while computing procurement orders."),
+        'message': fields.char('Latest error', size=128, help="Exception occurred while computing procurement orders."),
         'state': fields.selection([
             ('draft','Draft'),
             ('confirmed','Confirmed'),

