=== modified file 'account/account_invoice.py'
--- account/account_invoice.py	2013-03-29 15:59:38 +0000
+++ account/account_invoice.py	2013-04-09 18:18:18 +0000
@@ -1672,6 +1672,7 @@
                 val['manual'] = False
                 val['sequence'] = tax['sequence']
                 val['base'] = cur_obj.round(cr, uid, cur, tax['price_unit'] * line['quantity'])
+                val['tax_id']=tax['id']
 
                 if inv.type in ('out_invoice','in_invoice'):
                     val['base_code_id'] = tax['base_code_id']

