=== modified file 'account_voucher/account_voucher.py'
--- account_voucher/account_voucher.py	2012-08-28 10:58:11 +0000
+++ account_voucher/account_voucher.py	2012-12-17 22:04:34 +0000
@@ -1010,6 +1010,8 @@
         move_line_obj = self.pool.get('account.move.line')
         currency_obj = self.pool.get('res.currency')
         tax_obj = self.pool.get('account.tax')
+        voucher_line_obj = self.pool.get('account.voucher.line')
+        user_obj = self.pool.get('res.users')
         tot_line = line_total
         rec_lst_ids = []
 
@@ -1027,7 +1029,19 @@
             if line.amount == line.amount_unreconciled:
                 currency_rate_difference = line.move_line_id.amount_residual - amount
             else:
-                currency_rate_difference = 0.0
+                line_id = line.id
+                move_line_id = voucher_line_obj.browse(cr, uid, line.id, context=context).move_line_id.id
+                date_move = move_line_obj.browse(cr, uid, move_line_id).date
+                ctx = context.copy()
+                ctx.update({'date' : date_move})
+                company_user = user_obj.browse(cr, uid, uid).company_id
+                amount_invoice = currency_obj.compute(cr, uid, line.move_line_id.currency_id.id, company_user.currency_id.id, amount, context=ctx)
+                rate_invoice = amount_invoice / amount
+                currency_rate_difference = (rate_invoice - voucher_brw.payment_rate) * (amount/voucher_brw.payment_rate)
             move_line = {
                 'journal_id': voucher_brw.journal_id.id,
                 'period_id': voucher_brw.period_id.id,

