=== modified file 'mrp/mrp.py'
--- mrp/mrp.py	2012-10-05 10:44:47 +0000
+++ mrp/mrp.py	2012-10-16 17:33:24 +0000
@@ -899,6 +899,9 @@
         source_location_id = production.location_src_id.id
         if not destination_location_id:
             destination_location_id = source_location_id
+        analytic_acc_rm=False
+        if self._columns.get('analytic_acc_rm'):
+            analytic_acc_rm=production.analytic_acc_rm and production.analytic_acc_rm.id or False
         return stock_move.create(cr, uid, {
                         'name': move_name,
                         'picking_id': shipment_id,
@@ -913,6 +916,7 @@
                         'location_dest_id': destination_location_id,
                         'state': 'waiting',
                         'company_id': production.company_id.id,
+                        'analytic_acc': analytic_acc_rm,
                 })
           
     def _make_production_internal_shipment(self, cr, uid, production, context=None):
@@ -951,6 +955,9 @@
         source_location_id = production.product_id.product_tmpl_id.property_stock_production.id
         destination_location_id = production.location_dest_id.id
         move_name = _('PROD: %s') + production.name 
+        analytic_acc_fg=False
+        if self._columns.get('analytic_acc_fg'):
+            analytic_acc_fg=int(production.analytic_acc_fg and production.analytic_acc_fg.id or False)
         data = {
             'name': move_name,
             'date': production.date_planned,
@@ -964,6 +971,7 @@
             'move_dest_id': production.move_prod_id.id,
             'state': 'waiting',
             'company_id': production.company_id.id,
+            'analytic_acc': analytic_acc_fg,
         }
         move_id = stock_move.create(cr, uid, data, context=context)
         production.write({'move_created_ids': [(6, 0, [move_id])]}, context=context)
@@ -979,6 +987,9 @@
         destination_location_id = production.product_id.product_tmpl_id.property_stock_production.id
         if not source_location_id:
             source_location_id = production.location_src_id.id
+        analytic_acc_rm=False
+        if self._columns.get('analytic_acc_rm'):
+            analytic_acc_rm=production.analytic_acc_rm and production.analytic_acc_rm.id or False,
         move_id = stock_move.create(cr, uid, {
             'name': move_name,
             'date': production.date_planned,
@@ -992,6 +1003,7 @@
             'move_dest_id': parent_move_id,
             'state': 'waiting',
             'company_id': production.company_id.id,
+            'analytic_acc': analytic_acc_rm
         })
         production.write({'move_lines': [(4, move_id)]}, context=context)
         return move_id

