=== modified file 'stock/stock.py'
--- stock/stock.py	2012-10-05 07:47:57 +0000
+++ stock/stock.py	2012-10-16 17:30:45 +0000
@@ -2268,6 +2268,9 @@
         """
         # prepare default values considering that the destination accounts have the reference_currency_id as their main currency
         partner_id = (move.picking_id.address_id and move.picking_id.address_id.partner_id and move.picking_id.address_id.partner_id.id) or False
+        analytic_acc=False
+        if self.pool.get('stock.move')._columns.get('analytic_acc'):
+            analytic_acc=move.analytic_acc and move.analytic_acc.id or False,
         debit_line_vals = {
                     'name': move.name,
                     'product_id': move.product_id and move.product_id.id or False,
@@ -2277,6 +2280,7 @@
                     'partner_id': partner_id,
                     'debit': reference_amount,
                     'account_id': dest_account_id,
+                    'analytic_account_id': analytic_acc,
         }
         credit_line_vals = {
                     'name': move.name,
@@ -2287,6 +2291,7 @@
                     'partner_id': partner_id,
                     'credit': reference_amount,
                     'account_id': src_account_id,
+                    'analytic_account_id': analytic_acc
         }
 
         # if we are posting to accounts in a different currency, provide correct values in both currencies correctly
